src/synapse/workload_configuration.py

```diff
@@ -265,6 +265,27 @@ def enable_redis(current_yaml: dict, charm_state: CharmState) -> None:
         raise WorkloadError(str(exc)) from exc
 
 
+def enable_registration_secrets(current_yaml: dict, charm_state: CharmState) -> None:
+    """Change the Synapse configuration to enable registration secrets.
+
+    Args:
+        current_yaml: current configuration.
+        charm_state: Instance of CharmState.
+
+    Raises:
+        WorkloadError: something went wrong enabling registration secrets.
+    """
+    try:
+        if charm_state.registration_secrets is None:
```

Comment:
Could this be simplified to `if not charm_state.registration_secrets`? checking for None is usually for `null` type of checks where it is purposely left empty to carry meaning.

Reply:
Most of the code we have checks like this so I'm afraid that changing this one would be confusing. But is a good point for a refactoring that can be done in a different PR, WDYT

Reply:
ISD-2686 created to address this comment.